api/app/controllers/spree/api/v2/platform/resource_controller.rb

```diff
@@ -43,6 +45,16 @@ def destroy
 
           protected
 
+          def ensure_current_store(resource)
```

Comment:
@MatthewKennedy I think this requires a separate PR + more testing, also this is the same for admin, so maybe extracting it into a concern?

Reply:
@damianlegawiec sure, makes sense.